File: src/main/java/com/hedvig/claims/events/PaymentAddedEvent.java
package com.hedvig.claims.events;

import com.hedvig.claims.query.Carrier;
import lombok.Value;
import org.axonframework.serialization.Revision;

@Value
@Revision("3.0")
public class PaymentAddedEvent {
    String id;
    String claimsId;
    String userId;

    Double amount;
    Double deductible;
    String note;
    Boolean exGratia;
    String handlerReference;
    Carrier carrier;
}

File: src/main/java/com/hedvig/claims/events/PaymentAddedEvent.kt
package com.hedvig.claims.events

import java.time.LocalDateTime
import lombok.Value
import org.axonframework.serialization.Revision

@Revision("1.0")
data class PaymentAddedEvent (
  val id: String,
  val claimsId: String,
  val date: LocalDateTime,
  val userId: String,

  val amount: Double,
  val deductible: Double,
  val note: String,
  val payoutDate: LocalDateTime,
  val exGratia: Boolean,
  val handlerReference: String
)
